onnxruntime/core/providers/stvm/stvm_execution_provider.cc

```diff
@@ -0,0 +1,538 @@
+      Ort::CustomOpApi ort{*api};
+
+      std::vector<size_t> inds;
+      std::vector<DLTensor> dl_tensors_inputs;
+      for (auto& info : inputs_info_) {
+        // TODO(vvchernov): decomposition declaration only available with -std=c++1z or -std=gnu++1z
+        auto& i = info.first;
+        auto& shape = info.second;
+        const OrtValue* input_tensor = ort.KernelContext_GetInput(context, i);
+        ORT_ENFORCE(input_tensor->IsTensor());
+        const Tensor& tensor = input_tensor->Get<Tensor>();
+        const OrtDevice& device = tensor.Location().device;
+        auto tensor_info = ort.GetTensorTypeAndShape(input_tensor);
+        auto tensor_type = ort.GetTensorElementType(tensor_info);
+        if (!update_output_shapes_) {
+          std::vector<int64_t> ort_shape = ort.GetTensorShape(tensor_info);
+          ORT_ENFORCE(compare_shapes(shape, ort_shape));
+        }
+        ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+
+        DLTensor t;
+        t.device = GetDLDevice(device);
+        t.dtype = GetDataType(tensor_type);
+        t.strides = nullptr;
+        t.byte_offset = 0;
+        t.data = const_cast<void*>(ort.GetTensorData<void>(input_tensor));
+        t.ndim = shape.size();
+        t.shape = shape.data();
+        dl_tensors_inputs.push_back(t);
+        inds.push_back(i);
+      }
+      stvm::TVMSetInputs(*mod_, inds, dl_tensors_inputs);
+
+      size_t num_outputs = tensors_outputs_.size();
+      for (auto i = 0u; i < num_outputs; i++) {
+        //setup output tensor property
+        OrtValue* output_tensor = ort.KernelContext_GetOutput(context, i, output_shapes_[i].data(), output_shapes_[i].size());
+        ORT_ENFORCE(output_tensor->IsTensor());
+        const Tensor& tensor = output_tensor->Get<Tensor>();
+        const OrtDevice& device = tensor.Location().device;
+        auto tensor_info = ort.GetTensorTypeAndShape(output_tensor);
+        auto tensor_type = ort.GetTensorElementType(tensor_info);
+        ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
+
+        tensors_outputs_[i].device = GetDLDevice(device);
+        tensors_outputs_[i].dtype = GetDataType(tensor_type);
+        tensors_outputs_[i].data = ort.GetTensorMutableData<void>(output_tensor);
+      }
+
+      tvm::runtime::TVMRetValue rvalue;
+      stvm::TVMRun(*mod_, tensors_outputs_, &rvalue);
+
+      return Status::OK();
+    }
+  private:
+    void getTensorInfo(const TensorShapeProto& shape_proto,
+                       TVMTensorShape& ishape,
+                       size_t indx) {
+      TensorShape ort_shape = utils::GetTensorShapeFromTensorShapeProto(shape_proto);
+      int dims = ort_shape.NumDimensions();
+
+      ishape.resize(dims);
+      for (int j = 0; j < dims; ++j) {
+        int64_t dim = int64_t(ort_shape[j]);
+        if (dim > 0) {
+          ishape[j] = dim;
+        } else {
+          std::cout << "WARNING: input dimension is not positive value (dim = " << dim << "). ";
```

Comment:
use default logger instead of std::cout ?
LOGS_DEFAULT(WARNING) << "input dimension ...";

Reply:
Thanks for tip. I've redone it for ORT logger.